### PR TITLE
Allow approvers to impersonate the Cloud Build user-specified Service Account.

### LINKED ISSUE #1064: Grant roles/iam.serviceAccountUser for cloudbuild-sa to the build editors group.
If a user is part of the build_editors group and triggers a build by clicking on the RUN buttons, they would get the following error:

```
Failed to trigger build: generic::permission_denied: user does not have impersonation permission on the trigger service account specified: projects/PROJECT_ID/serviceAccounts/cloudbuild-sa@PROJECT_ID.iam.gserviceaccount.com
```

To fix this, 
`serviceAccountUser` role for `cloudbuild-sa` should be granted to the build_editors group in the CICD recipe.

### PR DESCRIPTION
**Pull request by pasha-gh**
Fix #1064

### CODE REVIEW

**Comment by pasha-gh** [comment]
Please ignore the long commit history left from previous changes.

File: templates/tfengine/components/cicd/main.tf [hunk 1]
```diff
@@ -134,6 +134,22 @@ resource "google_project_iam_member" "cloudbuild_builds_editors" {
...
 }
+
+# IAM permission to allow approvers to trigger builds.
+resource "google_service_account_iam_member" "cloudbuild_builds_editors" {
+  for_each = toset([
+    {{- range .build_editors}}
+    "{{.}}",
+    {{- end}}
+  ])
+  service_account_id = google_service_account.cloudbuild_sa.name
```
Comment (xingao267) [inline templates/tfengine/components/cicd/main.tf:145]: Should use `local.cloudbuild_sa_email` here instead.

File: templates/tfengine/components/cicd/main.tf [hunk 2]
```diff
@@ -134,6 +134,22 @@ resource "google_project_iam_member" "cloudbuild_builds_editors" {
     google_project_service.services,
   ]
 }
+
+# IAM permission to allow approvers to trigger builds.
```
Comment (xingao267) [inline templates/tfengine/components/cicd/main.tf:138]: IAM permission to allow approvers to impersonate the Cloud Build user-specified Service Account.

File: templates/tfengine/components/cicd/main.tf [hunk 3]
```diff
@@ -134,6 +134,22 @@ resource "google_project_iam_member" "cloudbuild_builds_editors" {
...
+    {{- range .build_editors}}
+    "{{.}}",
+    {{- end}}
+  ])
+  service_account_id = google_service_account.cloudbuild_sa.name
+  role               = "roles/iam.serviceAccountUser"
+  member             = each.value
+  depends_on = [
+    google_project_service.services,
+    google_service_account.cloudbuild_sa
```
Comment (xingao267) [inline templates/tfengine/components/cicd/main.tf:150]: not needed

**Comment by xingao267** [comment]
The PR title is not descriptive. Please update.

**Comment by xingao267** [comment]
@thomasfung-hk please take a look as well.

File: templates/tfengine/components/cicd/main.tf (hunk 1 above)
Comment (pasha-gh) [reply templates/tfengine/components/cicd/main.tf]: Fixed

File: templates/tfengine/components/cicd/main.tf (hunk 2 above)
Comment (pasha-gh) [reply templates/tfengine/components/cicd/main.tf]: Updated.

File: templates/tfengine/components/cicd/main.tf (hunk 3 above)
Comment (pasha-gh) [reply templates/tfengine/components/cicd/main.tf]: done.

**Comment by xingao267** [comment]
Have you tested this?

**Comment by xingao267** [comment]
LGTM as well. I'll approve for merging once it's tested and verified.

**Review by thomasfung-hk — APPROVED** [review]

**Comment by thomasfung-hk** [comment]
Just realized that the integration test hasn't been run; should that be done first?

**Comment by pasha-gh** [comment]
> Have you tested this?

Yes, I did test it with `google_service_account.cloudbuild_sa.name` and confirmed that build_editors have role/serviceAccount.user. We shouldn't have changed it to the email since `service_account_id` doesn't accept it.
Updated the PR and added `google_service_account.cloudbuild_sa.name` to the list of locals

File: templates/tfengine/components/cicd/main.tf [hunk 4]
```diff
@@ -41,8 +41,10 @@ data "google_project" "devops" {
 locals {
 {{- if get .service_account "exists" false}}
   cloudbuild_sa_email = "${var.service_account}@${var.project_id}.iam.gserviceaccount.com"
+  cloudbuild_sa_name = "projects/${var.project_id}/serviceAccounts/${var.service_account}@${var.project_id}.iam.gserviceaccount.com"
 {{- else}}
   cloudbuild_sa_email = google_service_account.cloudbuild_sa.email
+  cloudbuild_sa_name = google_service_account.cloudbuild_sa.name
```
Comment (xingao267) [inline templates/tfengine/components/cicd/main.tf:47]: Is `cloudbuild_sa_id` more accurate here?

**Review by xingao267 — APPROVED** [review]

**Pull request merged**